src/shared/modules/state-and-params.ts

```diff
@@ -168,3 +227,76 @@ export function getItemsStateAndParams({
         ...result,
     };
 }
+
+export function pickActionParamsFromParams(
+    params: ItemStateAndParams['params'],
+    returnWithPrefix?: boolean,
+) {
+    if (!params || isEmpty(params)) {
+        return {};
+    }
+
+    const actionParams: StringParams = {};
+    for (const [key, val] of Object.entries(params)) {
+        // starts with actionParams prefix (from'_ap_')
+        if (key.indexOf(ACTION_PARAM_PREFIX) === 0) {
+            const paramName = returnWithPrefix ? key : key.substr(ACTION_PARAM_PREFIX.length);
+            actionParams[paramName] = val;
+        }
+    }
+    return actionParams;
+}
+
+/**
+ * public function for getting params from object without actionParams
+ * @param params
+ */
+export function pickExceptActionParamsFromParams(params: ItemStateAndParams['params']) {
+    if (!params || isEmpty(params)) {
```

Comment:
There is `Object.entries(params))` below in code. And typescript throw error that `params` can have `undefined` value, but it doesn't because of `isEmpty(params)` check. TS can't evaluate type with `isEmpty` and etc. functions